src/libtiled/maprenderer.cpp

```diff
@@ -335,7 +350,17 @@ void CellRenderer::render(const Cell &cell, const QPointF &pos, const QSizeF &si
     const QRectF source(0, 0, fragment.width, fragment.height);
 
     mPainter->setTransform(transform);
-    mPainter->drawPixmap(target, image, source);
+    QPixmap tintedimg = image;
+    if(layer){
+        QColor tintColor = layer->tintColor();
+        if(tintColor != QColor(255, 255, 255, 255)){
+            tintedimg = QPixmap::fromImage(tinted(QImage(image.toImage())
```

Comment:
Instead of passing `QImage(image.toImage())` here, please take `QPixmap` in the `tinted` function and use `drawPixmap` there instead of `drawImage`. I doubt it matters for performance, but it's less code.

Reply:
done